DOCSP-39717 Transactions Page (#74)
mongodb/docs-java-rs: source/write/transactions.txt

.. _java-rs-transactions:

============
Transactions
============

.. contents:: On this page
   :local:
   :backlinks: none

>    :depth: 1

Reviewer: We have the Sample Data subsection on this page that would benefit from appearing in the ToC

```suggestion
   :depth: 2
```

   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: ACID, write, consistency, code example

Overview
--------

In this guide, you can learn how to use the {+driver-short+} to perform
**transactions**. Transactions allow you to run a series of operations that do
not apply until all data changes are successful. If any operation in the
transaction fails, the driver cancels the transaction and discards all data
changes without ever becoming visible.

In MongoDB, transactions run within logical **sessions**. A
session is a grouping of related
read or write operations that you intend to run sequentially. With sessions, you can
enable :manual:`causal consistency
</core/read-isolation-consistency-recency/#causal-consistency>` for a

> group of operations, and run :website:`ACID transactions

Reviewer: Nit:

```suggestion
group of operations as well as run :website:`ACID transactions
```

Author: Vale doesn't like "as well as"

         a transaction in progress for the session. To learn more about
         this method, see the :manual:`startTransaction() page
         </reference/method/Session.startTransaction/>`  in the {+mdb-server+} manual.
 
   * - ``abortTransaction()``
     - | Ends the active transaction for this session. Throws an exception
         if there is no active transaction for the session or if the
         transaction is already committed or ended. To learn more about
         this method, see the :manual:`abortTransaction() page
         </reference/method/Session.abortTransaction/>`  in the {+mdb-server+} manual.
 
   * - ``commitTransaction()``
     - | Commits the active transaction for this session. Throws an exception
         if there is no active transaction for the session or if the
         transaction was ended. To learn more about
         this method, see the :manual:`commitTransaction() page
         </reference/method/Session.commitTransaction/>`  in the {+mdb-server+} manual.

Transaction Example
-------------------

The following example demonstrates how to create a session, create a
transaction, and insert documents into multiple collections in one transaction.

> The code completes the following steps:

Reviewer: ```suggestion
The code executes the following steps:
```

- Creates a session from the client by using the ``startSession()`` method
- Starts a transaction by using the ``startTransaction()`` method
- Inserts documents into the ``restaurants`` and ``movies`` collections

> - Commits the transaction by using the ``commitTransaction()`` method

Reviewer: Suggestion: Switch to a numbered list here

.. literalinclude:: /includes/write/transactions/transactions.java
   :language: java
   :copyable:

>    :emphasize-lines: 26, 29, 33-36, 38

Reviewer: Suggestion: Trim some of the code sample – I think the most helpful part to include is everything contained within the `try` block.
